apps/ledger-live-mobile/src/screens/PostOnboarding/PostOnboardingHub.tsx

```diff
@@ -0,0 +1,194 @@
+import React, { useCallback, useEffect, useState } from "react";
+import { Box, Flex, Icons, Log, Notification, Text } from "@ledgerhq/native-ui";
+import { useTranslation } from "react-i18next";
+import { ScrollView, StyleSheet } from "react-native";
+import { SafeAreaView } from "react-native-safe-area-context";
+import styled from "styled-components/native";
+import { useFocusEffect, useNavigation } from "@react-navigation/native";
+import Animated, {
+  interpolate,
+  runOnJS,
+  useAnimatedStyle,
+  useSharedValue,
+  withDelay,
+  withTiming,
+} from "react-native-reanimated";
+import { StackScreenProps } from "@react-navigation/stack";
+import {
+  useAllPostOnboardingActionsCompleted,
+  useClearLastActionCompletedCallback,
+  usePostOnboardingHubState,
+} from "../../logic/postOnboarding/hooks";
+import PostOnboardingActionRow from "../../components/PostOnboarding/PostOnboardingActionRow";
+import { NavigatorName } from "../../const";
+
+const SafeContainer = styled(SafeAreaView).attrs({
+  edges: ["left", "bottom", "right"],
+})`
+  flex: 1;
+`;
+
+const Divider = styled(Box).attrs({
+  height: 0.5,
+  backgroundColor: "neutral.c30",
+  width: "100%",
+})``;
+
+const AnimatedFlex = Animated.createAnimatedComponent(Flex);
+
+const PostOnboardingHub: React.FC<StackScreenProps<{}>> = ({ navigation }) => {
+  const { t } = useTranslation();
+  const { lastActionCompleted, actionsState } = usePostOnboardingHubState();
+  const [popupOpened, setPopupOpened] = useState(true);
+  const { actionCompletedHubTitle, actionCompletedPopupLabel } =
+    lastActionCompleted ?? {};
+
+  const clearLastActionCompleted = useClearLastActionCompletedCallback();
+
+  useEffect(
+    () => () => {
+      /**
+       * The last action context (specific title & popup) should only be visible
+       * the 1st time the hub is navigated to after that action was completed.
+       * So here we clear the last action completed.
+       * */
+      clearLastActionCompleted();
+    },
+    [clearLastActionCompleted],
+  );
+
+  useEffect(() => {
+    /**
+     * Necessary because this component doesn't necessarily unmount when
+     * navigating to another screen, so we need to open the popup again in case
+     * the user closed it and then completed a new action.
+     */
+    setPopupOpened(true);
+  }, [actionCompletedPopupLabel]);
+
+  const navigateToMainScreen = useCallback(() => {
+    navigation.replace(NavigatorName.Base, {
+      screen: NavigatorName.Main,
+    });
+  }, [navigation]);
+
+  const handleClosePopup = useCallback(() => {
+    setPopupOpened(false);
+  }, [setPopupOpened]);
+
+  const animDoneValue = useSharedValue(0);
+
+  const allDone = useAllPostOnboardingActionsCompleted();
+
+  const triggerEndAnimation = useCallback(() => {
+    let dead = false;
+    let timeout: NodeJS.Timeout;
```

Comment:
We are on LLM, seems weird that the type is NodeJS.Timeout, why not `ReturnType<typeof setTimeout>`

Reply:
Ah great! Indeed that was a shitty typing, I didn't know about this `ReturnType` operator so I just used what my editor highlighted when I hover over `setTimeout` 😅 Thanks for the tip